File: app/src/main/java/com/caisl/loadinghelper/sample/adapter/TitleConfig.java
package com.caisl.loadinghelper.sample.adapter;


import android.view.View;

/**
 * @author caisl
 * @since 2019/4/9
 */
public class TitleConfig {

  private String mTitleText;
  private Type mType;
  private String mRightText;
  private View.OnClickListener mRightClickListener;

  public TitleConfig() {
  }

  public TitleConfig(String titleText, Type type) {
    mTitleText = titleText;
    mType = type;
  }

  public TitleConfig(String rightText, View.OnClickListener rightClickListener) {
    mRightText = rightText;
    mRightClickListener = rightClickListener;
  }

  public String getTitleText() {
    return mTitleText;
  }

  public void setTitleText(String titleText) {
    mTitleText = titleText;
  }

  public Type getType() {
    return mType;
  }

  public void setType(Type type) {
    mType = type;
  }

  public String getRightText() {
    return mRightText;
  }

  public void setRightText(String rightText) {
    mRightText = rightText;
  }

  public View.OnClickListener getRightClickListener() {
    return mRightClickListener;
  }

  public void setRightClickListener(View.OnClickListener rightClickListener) {
    mRightClickListener = rightClickListener;
  }

  public enum Type {
    BACK, NO_BACK
  }

}

File: app/src/main/java/com/dylanc/loadinghelper/sample/base/TitleConfig.kt
package com.dylanc.loadinghelper.sample.base


import android.view.View

/**
 * @author Dylan Cai
 * @since 2019/4/9
 */
open class TitleConfig(
  var titleText: String,
  var type: Type
) {
  var rightIcon: Int = 0
    private set
  var rightText: String? = null
    private set
  var onRightBtnClickListener: View.OnClickListener? = null
    private set

  fun setRightBtn(rightIcon: Int, onRightBtnClickListener: View.OnClickListener?) {
    this.rightIcon = rightIcon
    this.onRightBtnClickListener = onRightBtnClickListener
  }

  fun setRightBtn(rightText: String, onRightBtnClickListener: View.OnClickListener?) {
    this.rightText = rightText
    this.onRightBtnClickListener = onRightBtnClickListener
  }

  enum class Type {
    BACK, NO_BACK
  }
}
